### PR TITLE
[uikit] Update for Xcode 9 - Part 6

### PR DESCRIPTION
**Pull request by VincentDondain**
NSAttributedString.h
NSFileProviderExtension.h
NSLayoutAnchor.h
UIAccessibility.h
UIAccessibilityAdditions.h
UIAccessibilityContainer.h
UIAccessibilityContentSizeCategoryImageAdjusting.h
UIAccessibilityCustomRotor.h
UIAccessibilityLocationDescriptor.h
UITabBar.h
UINavigationBar.h
UIScreen.h

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -450,18 +451,50 @@ interface NSLayoutAnchor<AnchorType>
...
 	[NoWatch]
 	[iOS (9,0)]
+	[TV (10,0)]
 	[BaseType (typeof(NSLayoutAnchor<NSLayoutXAxisAnchor>))]
 	[DisableDefaultCtor] // Handle is nil
 	interface NSLayoutXAxisAnchor
 	{
+		[iOS (10,0)]
+		[Export ("anchorWithOffsetToAnchor:")]
+		NSLayoutDimension AnchorWithOffset (NSLayoutXAxisAnchor otherAnchor);
```
Comment (spouliot) [inline src/uikit.cs]: lack of verb/action in the name -> prefix with `Create`

File: src/uikit.cs [hunk 2]
```diff
@@ -450,18 +451,50 @@ interface NSLayoutAnchor<AnchorType>
...
 	[NoWatch]
 	[iOS (9,0)]
+	[TV (10,0)]
 	[BaseType (typeof(NSLayoutAnchor<NSLayoutYAxisAnchor>))]
 	[DisableDefaultCtor] // Handle is nil
 	interface NSLayoutYAxisAnchor
 	{
+		[iOS (10,0)]
+		[Export ("anchorWithOffsetToAnchor:")]
+		NSLayoutDimension AnchorWithOffset (NSLayoutYAxisAnchor otherAnchor);
```
Comment (spouliot) [inline src/uikit.cs]: same, prefix with `Create`

File: src/uikit.cs [hunk 3]
```diff
@@ -1697,10 +1818,18 @@ partial interface UIAccessibilityCustomAction {
 	    [Export ("initWithName:target:selector:")]
 	    IntPtr Constructor (string name, NSObject target, Selector selector);
 	
+	[TV (11,0), iOS (11,0)]
+	[Export ("initWithAttributedName:target:selector:")]
+	IntPtr Constructor (NSAttributedString attributedName, [NullAllowed] NSObject target, Selector selector);
+
```
Comment (spouliot) [inline src/uikit.cs]: minor: indent looks incorrect (single tab)

File: src/uikit.cs [hunk 4]
```diff
@@ -1697,10 +1818,18 @@ partial interface UIAccessibilityCustomAction {
...
 		[NullAllowed] // by default this property is null
 	    [Export ("name")]
 	    string Name { get; set; }
 	
+	[TV (11,0), iOS (11,0)]
+	[Export ("attributedName", ArgumentSemantic.Copy)]
+	NSAttributedString AttributedName { get; set; }
+
```
Comment (spouliot) [inline src/uikit.cs]: minor: indent looks incorrect (single tab)

File: src/uikit.cs [hunk 5]
```diff
@@ -1688,6 +1764,51 @@ interface UIAccessibilityContainer {
...
+		[iOS (11,0), TV (11,0)]
+		[Export ("accessibilityContainerType", ArgumentSemantic.Assign)]
+		UIAccessibilityContainerType AccessibilityContainerType { get; set; }
+	}
+
+	interface IUIAccessibilityContainerDataTableCell {}
+
+	[iOS (11,0), TV (11,0)]
+	[Protocol, Model]
```
Comment (spouliot) [inline src/uikit.cs]: not clear if a `[Model]` is needed, is this type user created ?

File: src/uikit.cs [hunk 6]
```diff
@@ -1851,6 +1996,54 @@ interface UIAccessibilityAction {
 		UIAccessibilityCustomAction [] AccessibilityCustomActions { get; set; }
 	}
 
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	interface UIAccessibilityDragging {
```
Comment (spouliot) [inline src/uikit.cs]: looks weird, no `[BaseType]`, no `[Protocol]` ?

File: src/uikit.cs [hunk 7]
```diff
@@ -1851,6 +1996,54 @@ interface UIAccessibilityAction {
...
+		[Export ("name", ArgumentSemantic.Strong)]
+		string Name { get; }
+
+		[Export ("attributedName", ArgumentSemantic.Strong)]
+		NSAttributedString AttributedName { get; }
+	}
+
+	[NoWatch]
+	[TV (11,0), iOS (11,0)]
+	[Protocol, Model]
```
Comment (spouliot) [inline src/uikit.cs]: the `[Model]` does not seem usable as a base type

File: src/uikit.cs [hunk 8]
```diff
@@ -17999,6 +18260,116 @@ interface UIPasteConfigurationSupporting {
...
+
+		[Export ("additionalLeadingNavigationBarButtonItems", ArgumentSemantic.Strong)]
+		UIBarButtonItem[] AdditionalLeadingNavigationBarButtonItems { get; set; }
+
+		[Export ("additionalTrailingNavigationBarButtonItems", ArgumentSemantic.Strong)]
+		UIBarButtonItem[] AdditionalTrailingNavigationBarButtonItems { get; set; }
+
+		[Async]
+		[Export ("revealDocumentAtURL:importIfNeeded:completion:")]
+		void RevealDocumentAtUrl (NSUrl url, bool importIfNeeded, [NullAllowed] Action<NSUrl, NSError> completion);
```
Comment (spouliot) [inline src/uikit.cs]: remove `AtUrl`, it's implied

File: src/uikit.cs [hunk 9]
```diff
@@ -17999,6 +18260,116 @@ interface UIPasteConfigurationSupporting {
...
+
+		[Async]
+		[Export ("importDocumentAtURL:nextToDocumentAtURL:mode:completionHandler:")]
+		void ImportDocumentAtUrl (NSUrl documentUrl, NSUrl neighbourUrl, UIDocumentBrowserImportMode importMode, Action<NSUrl, NSError> completion);
```
Comment (spouliot) [inline src/uikit.cs]: remove `AtUrl`, it's implied

File: src/uikit.cs [hunk 10]
```diff
@@ -1688,6 +1764,51 @@ interface UIAccessibilityContainer {
...
+	[BaseType (typeof(NSObject))]
+	interface UIAccessibilityContainerDataTable {
+		[Abstract]
+		[Export ("accessibilityDataTableCellElementForRow:column:")]
+		[return: NullAllowed]
+		IUIAccessibilityContainerDataTableCell GetAccessibilityDataTableCellElement (nuint row, nuint column);
+
+		[Abstract]
+		[Export ("accessibilityRowCount")]
+		nuint GetAccessibilityRowCount ();
```
Comment (rolfbjarne) [inline src/uikit.cs]: Abstract protocol members can be properties.

File: src/uikit.cs [hunk 11]
```diff
@@ -1688,6 +1764,51 @@ interface UIAccessibilityContainer {
...
+
+		[Abstract]
+		[Export ("accessibilityColumnCount")]
+		nuint GetAccessibilityColumnCount ();
```
Comment (rolfbjarne) [inline src/uikit.cs]: Abstract protocol members can be properties.

File: src/uikit.cs [hunk 12]
```diff
@@ -17999,6 +18260,116 @@ interface UIPasteConfigurationSupporting {
...
+		[Async]
+		[Export ("revealDocumentAtURL:importIfNeeded:completion:")]
+		void RevealDocumentAtUrl (NSUrl url, bool importIfNeeded, [NullAllowed] Action<NSUrl, NSError> completion);
+
+		[Async]
+		[Export ("importDocumentAtURL:nextToDocumentAtURL:mode:completionHandler:")]
+		void ImportDocumentAtUrl (NSUrl documentUrl, NSUrl neighbourUrl, UIDocumentBrowserImportMode importMode, Action<NSUrl, NSError> completion);
+
+		[Export ("transitionControllerForDocumentURL:")]
+		UIDocumentBrowserTransitionController TransitionControllerForDocumentUrl (NSUrl documentUrl);
```
Comment (rolfbjarne) [inline src/uikit.cs]: I think `GetTransitionController` sounds better.

File: src/uikit.cs (hunk 5 above)
Comment (VincentDondain) [reply src/uikit.cs]: Yea it's not clear from the doc nor the headers, let's un-model it until we're proven it's required.

**Pull request merged**